lib/Quantization/Quantization.cpp

```diff
@@ -30,6 +30,105 @@ namespace {
 using namespace glow;
 using namespace glow::quantization;
 
+/// Support quantized Log by replacing it with an IntLookupTable with a new
+/// mapping given the input and output quantization parameters.
+static void replaceQuantizedLogWithLookupTable(Function &F, const LogNode &LN) {
+  TypeRef outTy = LN.getResult().getType();
+  TypeRef inTy = LN.getInput().getType();
+
+  auto inputRange = inTy->getQuantizedValueRange();
+  (void)inputRange;
+  assert(inputRange.first >= 0 &&
+         "Input range must not be negative since this is input to log().");
+
+  // Pass a function returning log here to create the mapping. Note that the
+  // interval is always extended to include zero, so we check if the input is
+  // zero and if so use log(float min), i.e. closest positive value to zero,
+  // as -inf is unsupported to convert to int.
+  auto logFun = [](float a) {
+    return (a == 0.0) ? log(std::numeric_limits<float>::min()) : log(a);
+  };
+  std::vector<int8_t> mapping =
+      glow::quantization::createMapping(inTy, outTy, logFun);
+
+  // Create a new int lookup table with this newly calculated mapping to
+  // implement this quantized log.
+  IntLookupTableNode *ILT = F.createIntLookupTable(
+      LN.getName().str() + ".log", LN.getInput(), mapping, outTy);
+
+  LN.getResult().replaceAllUsesOfWith(ILT);
+}
+
+/// Support quantized Tanh by replacing it with an IntLookupTable.
+static void replaceQuantizedTanhWithLookupTable(Function &F,
+                                                const TanhNode &TN) {
+  // Quantized tanh operator expects input to be in a certain floating point
+  // range. This operator works based on the precomputed table and has to
+  // process input in a range of [-3.0, 3.0]. Tanh asymptotically approaches
+  // +/-1.0 and is already +/-.995 at +/-3.0.
+  // The output quantization parameters are chosen to represent the floating
+  // point range of [-1.0, 1.0].
+  auto inputQuantizationParams =
+      glow::quantization::chooseQuantizationParams(-3.0, 3.0);
+  auto tanhInTy = F.getParent()->uniqueType(
+      ElemKind::Int8QTy, TN.getResult().dims(), inputQuantizationParams.scale,
+      inputQuantizationParams.offset);
+
+  // Make sure input is clipped in [-3.0, 3.0] floating point range.
+  auto *rescaleInputNode =
+      F.createRescaleQuantized(TN.getName(), TN.getInput(), tanhInTy);
+
+  // Make sure output is clipped in [-1.0, 1.0] floating point range.
+  auto outputQuantizationParams =
+      glow::quantization::chooseQuantizationParams(-1.0, 1.0);
+  auto resultOutTy = F.getParent()->uniqueType(
+      ElemKind::Int8QTy, rescaleInputNode->getResult().dims(),
+      outputQuantizationParams.scale, outputQuantizationParams.offset);
+
+  auto *quantizedNode =
+      F.createIntTanh(TN.getName(), rescaleInputNode, resultOutTy);
```

Comment:
nit: might be worth a comment that the lookup table is created in here (I had to go looking).